### PR TITLE
Alteration of requester on ticket update

### PR DESCRIPTION
**Pull request by btry**
When updating a ticket the requester of the associated issue is replaced by the author of the ticket.

internal ref 23560

### CODE REVIEW

File: hook.php [hunk 1]
```diff
@@ -374,6 +376,24 @@ function plugin_formcreator_hook_update_ticket(CommonDBTM $item) {
...
+   $requester = $DB->request([
+      'SELECT' => 'users_id',
+      'FROM' => Ticket_User::getTable(),
+      'WHERE' => [
+         'tickets_id' => $item->getID(),
+         'type' => CommonITILActor::REQUESTER,
+      ],
+      'ORDER' => ['id'],
+      'LIMIT' => '1',
+   ])->next();
```
Comment (AdrienClairembault) [inline hook.php:390]: `->next()` do not return anything in GLPI 10 so `$requester` will be undefined.
Maybe use `current()` so the code work on both versions ?

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]

File: hook.php (hunk 1 above)
Comment (btry) [reply hook.php:390]: If I remember wel calling current() without previous call to next() will not work. That'sd why I plan to to a specific version of  the fix for branch 2.13 of Formcreator, specifically for this suggestion.

Comment (AdrienClairembault) [reply hook.php:390]: You can even do something like `current($requesters)['users_id'] ?? 0;`

Comment (AdrienClairembault) [reply hook.php:390]: @cedric-anne Is this the best way to have the same code working for both versions ?

**Review by AdrienClairembault — APPROVED** [review]

File: hook.php (hunk 1 above)
Comment (cedric-anne) [reply hook.php:390]: maybe `current(iterator_to_array($iterator))`

**Pull request merged**